.github/workflows/ci.yml

```diff
@@ -1,8 +1,11 @@
 name: CI
 
+# only run tests for pull requests cause no file has to be changed without review
+# open -> open the pull request
+# synchronize -> push to branch of pull request
 on:
-  push:
-    branches: [ 'main', 'master', 'dev-*', 'fix-*' ]
+  pull_request:
```

Comment:
changed to pull_request cause all future changes will get in through a pull request